compiler/passes/codegen.cpp

```diff
@@ -262,6 +264,23 @@ genFtable(Vec<FnSymbol*> & fSymbols, bool isHeader) {
     if (fSymbols.n == 0)
       fprintf(hdrfile, "(chpl_fn_p)0");
     fprintf(hdrfile, "\n};\n");
+
+    // function information (names ...) ??? some flag to dump these?
+    fprintf(hdrfile, "chpl_fn_info %s[] = {\n", finfo_name);
+    first = true;
+
+    forv_Vec(FnSymbol, fn, fSymbols) {
+      if (!first)
+        fprintf(hdrfile, ",\n");
+      fprintf(hdrfile, "{\"%s\", %d, %d}", fn->cname,
+              getFilenameLookupPosition(fn->astloc.filename),
+              fn->astloc.lineno);
+      first = false;
+    }
+    if (!first)
+      fprintf(hdrfile, ",\n");
+    fprintf(hdrfile, "{(char *)0, 0, 0}\n};\n");
+
```

Comment:
I'd prefer this code be in a new function. Then, I'd prefer that this new function be called in codegen_header_compilation_config. To make that work, just move the for_vector computing ftableVec (around line 1107) further up in the codegen_header function and pass the resulting vector to codegen_header_compilation_config.